pkg/csi/service/cns/nodes.go

```diff
@@ -58,7 +58,20 @@ func (nodes *Nodes) nodeAdd(obj interface{}) {
 }
 
 func (nodes *Nodes) nodeUpdate(oldObj interface{}, newObj interface{}) {
-	nodes.nodeRegister(newObj)
+	newNode, ok := newObj.(*v1.Node)
+	if newNode == nil || !ok {
+		klog.Warningf("nodeUpdate: unrecognized object newObj %+v", newObj)
```

Comment:
This [won't log the full type](https://play.golang.org/p/3WV14R3hnjS). I recommend [the following](https://play.golang.org/p/IEXziLIokqz):

```go
klog.Warningf("nodeUpdate: unrecognized object newObj %[1]T%+[1]v", newObj)
```

Reply:
Done